languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/de-DE-AT/grammar.xml

```diff
@@ -235,6 +235,86 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
     </category>
 
     <category id="TYPOGRAPHY" name="Typografie" type="typographical">
+        <rulegroup id="ZAHL_FORMAT_PREMIUM" name="Er wiegt ungefähr 10000 (10.000) KG.">
+            <antipattern>
+                <token regexp="yes">\.|,|.*Nr\.?|No\.?|HRA|HRB|.*Nummer</token>
+                <token regexp="yes">(\d){5,30}</token>
+            </antipattern>
+            <rule><!--1-->
+                <pattern>
+                    <marker>
+                        <token regexp="yes">(\d){5,6}</token>
+                    </marker>
+                    <token regexp="yes" case_sensitive="yes">KG|G|Gramm|Kilo(gramm)?|KM|Kilometer|M|CM|DM|MM|((Milli|Centi|Deci)?meter)|T(onnen)?|((K|M|G|T|P)i?)?B(it|ytes?)?|Usern?|Zeichen|Sätzen?|Sekunden|Minuten|Stunden|Nutzern?|Schritten?|(\(See\)-)?Meilen|Wörtern?|Menschen|Tagen?|[£$€]|Euro?|Dollar|USD</token>
+                </pattern>
+                <message>Durch Punkte als Tausender-Trennzeichen lässt sich die Lesbarkeit verbessern.</message>
+                <suggestion><match no="1" regexp_match="^(.+)(\d{3})$" regexp_replace="$1.$2" /></suggestion>
+                <example correction="100.000">Hier werden bestimmt <marker>100000</marker> GB an Daten gespeichert.</example>
+                <example>1 € = 1,95583 DM.</example>
+            </rule>
+            <rule><!--2-->
+                <pattern>
+                    <marker>
+                        <token regexp="yes">(\d){7,9}</token>
+                    </marker>
+                    <token regexp="yes" case_sensitive="yes">KG|G|Gramm|Kilo(gramm)?|KM|Kilometer|M|CM|DM|MM|((Milli|Centi|Deci)?meter)|T(onnen)?|((K|M|G|T|P)i?)?B(it|ytes?)?|Usern?|Zeichen|Sätzen?|Sekunden|Minuten|Stunden|Nutzern?|Schritten?|(\(See\)-)?Meilen|Wörtern?|Menschen|Tagen?|[£$€]|Euro?|Dollar|USD</token>
+                </pattern>
+                <message>Durch Punkte als Tausender-Trennzeichen lässt sich die Lesbarkeit verbessern.</message>
+                <suggestion><match no="1" regexp_match="^(.+)(\d{3})(\d{3})$" regexp_replace="$1.$2.$3" /></suggestion>
+                <example correction="1.500.000">Hier werden bestimmt <marker>1500000</marker> GB an Daten gespeichert.</example>
+            </rule>
+            <rule><!--3-->
+                <pattern>
+                    <marker>
+                        <token regexp="yes">(\d){5,30}</token>
+                    </marker>
+                    <token regexp="yes" case_sensitive="yes">KG|G|Gramm|Kilo(gramm)?|KM|Kilometer|M|CM|DM|MM|((Milli|Centi|Deci)?meter)|T(onnen)?|((K|M|G|T|P)i?)?B(it|ytes?)?|Usern?|Zeichen|Sätzen?|Sekunden|Minuten|Stunden|Nutzern?|Schritten?|(\(See\)-)?Meilen|Wörtern?|Menschen|Tagen?|[£$€]|Euro?|Dollar|USD</token>
+                </pattern>
+                <message>Durch Punkte als Tausender-Trennzeichen lässt sich die Lesbarkeit verbessern.</message>
```

Comment:
Please track how common this is (opens) and if it's common, we should have a suggestion (e.g. with a rule filter).